kolibri_exercise_perseus_plugin/assets/src/vue/index.vue

```diff
@@ -1,151 +1,327 @@
+      this.itemRendererFactory = this.react.createFactory(this.perseus.ItemRenderer);
+    },
+
+    beforeDestroy() {
+      this.clearItemRenderer();
+      this.$emit('stopTracking');
+    },
+
+    destroyed() {
+      // Clean up the global namespace pollution that Perseus necessitates.
+      delete global.React;
+      delete global.$;
+      delete global.jQuery;
+      delete global.i18n;
+      delete global.KAS;
+      delete global.MathQuill;
+      delete global.ReactDOM;
+      delete global.Exercises;
+    },
+
+    $trNameSpace: 'perseus',
     $trs: {
-      goal: 'Try to get {count, number, integer} ' +
-        '{count, plural, one {check mark} other {check marks}} to show up',
-      tryAgain: 'Try again!',
+      showScratch: 'Show scratchpad',
+      notAvailable: 'The scratchpad is not available',
+      loading: 'Loading',
+      hint: 'Get a hint',
+      hintLabel: 'Hint:',
+      noMoreHint: 'No more hints',
+    },
+    components: {
+      'icon-button': require('kolibri.coreVue.components.iconButton'),
+    },
+    name: 'exercise-perseus-renderer',
+    props: {
+      scratchpad: {
+        type: Boolean,
+        default: false,
+      },
+      initialHintsVisible: {
+        type: Number,
+        default: 0,
+      },
+      problemNumber: {
+        type: Number,
+        default: 1,
+      },
+      defaultFile: {
+        type: Object,
+        required: true,
+      },
+      itemId: {
+        type: String,
+        required: true,
+      },
+      answerState: {
+        type: Object,
+        default: {},
+      },
     },
     data: () => ({
-      item: undefined,
-      items: undefined,
-      exercise: undefined,
-      itemId: undefined,
-      // how many questions need to get right?
-      passRatioM: 4,
-      // how many most recent questions taken into account?
-      passRatioN: 5,
-      waiting: true,
-      success: false,
-      correctness: true,
+      // Is the perseus item renderer loading?
+      loading: true,
+      // state about the answer
+      message: null,
+      // number of available hints
+      availableHints: 0,
+      // default item data
+      item: {},
     }),
-    computed: {
-      recentAttempts() {
-        if (!this.pastattempts) {
-          return [];
+
+    methods: {
+      validateItemData(obj) {
+        return [
+          // A somewhat protracted validator to ensure that our item data conforms
+          // to that expected by the Perseus ItemRenderer,
+          // c.f. https://github.com/Khan/perseus/blob/master/src/item-renderer.jsx#L35
+          'calculator',
+          'chi2Table',
+          'periodicTable',
+          'tTable',
+          'zTable',
+        ].reduce(
+            /* eslint-disable no-mixed-operators */
+            // Loop through all of the above properties and ensure that if the 'answerArea'
+            // property of the item has them, then their values are set to Booleans.
+            (prev, key) => !(!prev ||
```

Comment:
for this and the other fat arrow reducer, perhaps using named function syntax would help "self-document" what the reducer does?